### PR TITLE
Bug layer ordering gets reversed

### LINKED ISSUE #189: Bug: Layer ordering can end up in a reversed state
A bug surfaced that causes the property order to be set incorrectly. In normal operation, this bug was resolved by a double negative, that is, a situation where the ordered array was reversed, and then reversed again later.

However, the health check endpoint calls a getter which mutates the ordered array in place, causing this bug to occur.

The list of active sources is reversed by the [`Properties.sources()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L34) getter. That getter is called by the [`Sources.health()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/sources.js#L155) function used by both the `/v1/health` and `/v1/status` APIs.

Additionally, the list of layers is reversed every time [`Properties.build()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L117) is called.

### PR DESCRIPTION
**Pull request by dgreene-r7**
This PR refactors parts of the `Properties` and `View` classes to use immutable objects as data storage rather than plain JS objects. This fixes problems where certain built in methods were mutating objects in place.

It also pins the node version to the latest LTS/argon release and reorders the way the Consul, Metadata, and local Config layers are loaded.

Resolves #189.

### CODE REVIEW

**Comment by fmitchell-r7** [comment]
Thanks for fixing this!

Would you mind adding some tests in to make sure this bug stays fixed? Specifically, I'm thinking there should be tests for a) ensuring multiple calls to `Properties.build()` don't reverse the layer ordering, and b) ensuring calls to the `/v1/health` and `/v1/status` APIs don't reverse the active sources ordering.

**Comment by fmitchell-r7** [comment]
👍 Thanks for adding those tests! I'd get a 👍 from @jmanero-r7 as well to make sure there's nothing subtle we're missing here, then 🚢 it.

**Comment by rvanburen-r7** [comment]
👍

File: lib/properties.js [hunk 1]
```diff
@@ -13,8 +14,8 @@ class Properties extends EventEmitter {
 
     this.initialized = false;
 
-    this.layers = [];
-    this.properties = {};
+    this._layers = Immutable.List();
+    this._properties = Immutable.Map();
```
Comment (jmanero-r7) [inline lib/properties.js]: I don't think that introducing additional complexity is the right way to solve this. Rather, we should refactor the parts that actually (unnecessarily) require usage of `reverse` in the first place, making the model simpler and more consistent with the interfaces it's supposed to present.

Comment (dgreene-r7) [inline lib/properties.js]: `Immutable` is going to be introduced in #188 anyway. Not sure why we shouldn't take advantage of it.

**Comment by dgreene-r7** [comment]
Superseded by #196.

**Pull request closed without merging**